src/JsonSchema/Constraints/ObjectConstraint.php

```diff
@@ -103,10 +103,10 @@ public function validateElement($element, $matches, $objectDefinition = null, $p
                 $this->addError($path, "The presence of the property " . $i . " requires that " . $require . " also be present", 'requires');
             }
 
-            if (!$definition) {
+            //if (!$definition) {
                 // normal property verification
-                $this->checkUndefined($value, new \stdClass(), $path, $i);
-            }
+                //$this->checkUndefined($value, new \stdClass(), $path, $i);
```

Comment:
I really don't know why this was there.. what does the comment "`// normal property verification`" mean?

If we have no schema at all, is there any need to continue the recursion? If I check `UndefinedConstraint`, all it does is always checking stuff in `$schema` which was an empty `\stdClass()` all the way - anybody knows more?